module/programs/helper.js

```diff
@@ -1049,27 +1047,44 @@ module.exports = class ProgramsHelper {
               }
             }
             let consentResponse = await userService.setUserConsent(userToken, userConsentRequestBody)
-            console.log("consentResponse ",consentResponse)
 
             if(!consentResponse.success){
               throw {
                 message: constants.apiResponses.PROGRAM_JOIN_FAILED,
                 status: httpStatusCode.bad_request.status
               }
             }
+            
           }
-          pushProgramUsersDetailsToKafka = true; 
+          
+        }
+
+        // if requestForPIIConsent Is false and user not joined program till now then set pushProgramUsersDetailsToKafka = true; 
+        // if requestForPIIConsent == true and data.consentShared value is true which means user interacted with the consent popup set pushProgramUsersDetailsToKafka = true;
+        if((programData[0].hasOwnProperty('requestForPIIConsent') && programData[0].requestForPIIConsent === false && !programUsersDetails.length > 0) ||
+          ((programData[0].hasOwnProperty('requestForPIIConsent') && programData[0].requestForPIIConsent === true) && (data.hasOwnProperty('consentShared') && data.consentShared == true) ) ){
+            
+            pushProgramUsersDetailsToKafka = true; 
+            
+            // if programUsersDetails[0].consentShared === true which means the data is already pushed to Kafka once, So revert pushProgramUsersDetailsToKafka to false
+            if (programUsersDetails.length > 0 && programUsersDetails[0].hasOwnProperty('consentShared') && programUsersDetails[0].consentShared === true) {
+              pushProgramUsersDetailsToKafka = false;
```

Comment:
@VISHNUDAS-tunerlabs Logic can be by default pushProgramUsersDetailsToKafka = false; as defined Line 968. Then just set it to true if needed in the fucntion. In the same if block first setting to true and then again false doesn't seem to be a good way to do so

Reply:
@aks30 changes added